src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -16,39 +16,44 @@ Example g++ command
     g++ -g -std=c++14 -I$XILINX_XRT/include -L$XILINX_XRT/lib -o host.exe host.cpp -lxrt_coreutil -pthread
 
 
-The core data structures in C and C++ are as below
+The XRT native API supports both the C and C++ flavor of APIs. For general host code development, C++-based APIs are recommended, hence this document only describes the C++-based API interfaces. The full Doxygen generated C and C++ API documentation can be found in `<./xrt_native.main.rst>`_.
 
-+---------------+---------------+-------------------+
-|               |   C++ Class   |  C Type (Handle)  |
-+===============+===============+===================+
-|   Device      | xrt::device   |  xrtDeviceHandle  |
-+---------------+---------------+-------------------+
-|   XCLBIN      | xrt::xclbin   |  xrtXclbinHandle  |
-+---------------+---------------+-------------------+
-|   Buffer      | xrt::bo       |  xrtBufferHandle  |
-+---------------+---------------+-------------------+
-|   Kernel      | xrt::kernel   |  xrtKernelHandle  |
-+---------------+---------------+-------------------+
-|   Run         | xrt::run      |  xrtRunHandle     |
-+---------------+---------------+-------------------+
-|   Graph       | TBD           |  xrtGraphHandle   |
-+---------------+---------------+-------------------+
 
-All the core data structures are defined inside in the header files at ``$XILINX_XRT/include/experimental/`` directory. In the user host code, it is sufficient to include ``"experimental/xrt_kernel.h"`` and ``"experimental/xrt_aie.h"`` (when using Graph APIs) to access all the APIs related to these data structure.
+The C++ Class objects used for the APIs are 
+
++----------------------+-------------------+
+|                      |   C++ Class       |  
++======================+===================+
+|   Device             | ``xrt::device``   |  
++----------------------+-------------------+
+|   XCLBIN             | ``xrt::xclbin``   |  
++----------------------+-------------------+
+|   Buffer             | ``xrt::bo``       |  
++----------------------+-------------------+
+|   Kernel             | ``xrt::kernel``   |  
++----------------------+-------------------+
+|   Run                | ``xrt::run``      |  
++----------------------+-------------------+
+| User-managed Kernel  | ``xrt::ip``       |  
++----------------------+-------------------+
+|   Graph              | ``xrt::graph``    |  
++----------------------+-------------------+
+
+All the core data structures are defined inside in the header files at ``$XILINX_XRT/include/xrt/`` directory. Few newer features such as ``xrt::ip``, ``xrt::aie`` related header files are inside ``$XILINX_XRT/include/experimental`` directory. 
```

Comment:
I think you should extend the table above to show what and where the corresponding header is.    Then just make a note, that the APIs that are in the experimental folder are subject to **breaking** changes.

Reply:
done , and note about experimental is added